Rewrite conclusion (#48)
greenelab/text_mined_hetnet_manuscript: content/06.conclusion_future_direction.md

## Conclusion and Future Direction

Filling out knowledge bases via manual curation can be an arduous and erroneous task [@doi:10.1093/bioinformatics/btm229].

> As the rate of publications increases manual curation alone becomes an impractical approach.

Reviewer: ```suggestion
As the rate of publications increases, relying on manual curation alone becomes impractical.
```

Data programming, a paradigm that uses label functions as a means to speed up the annotation process, can be used as a solution for this problem.

> An obstabcle this paradigm faces is that creating a useful label function takes a considerable amount of time. 

Reviewer: ```suggestion
An obstacle for this paradigm is creating useful label functions, which takes a considerable amount of time. 
```

> We tested the feasibility of reusing label functions as a way to speed up the  label function creation process.

Reviewer: ```suggestion
We tested the feasibility of reusing label functions as a way to reduce the total number of label functions required for strong prediction performance.
```

> We conclude that label function can be re-used with closely related edge types (Compound binds Gene and Gene interacts Gene), but is practically futile for distantly related ones (Compound treats Disease and Disease associates Gene).

Reviewer: ```suggestion
We conclude that label functions may be re-used with closely related edge types, but that re-use does not improve performance for most pairings.
```

> More sophisticated methods may be able to connect distantly related edge types and avoid many of the drawbacks we encountered in this work.

Reviewer: ```suggestion
```

> The discriminative model's performance improves as more edge-specific label functions are incorporated into the generative model; however, we did notice that performance greatly depends on the generativce model.

Reviewer: ```suggestion
The discriminative model's performance improves as more edge-specific label functions are incorporated into the generative model; however, we did notice that performance greatly depends on the generative model.
```

> We recommend that future endeavors focus on efficiently tuning the generative model to ensure proper training of the discriminative model.

Reviewer: ```suggestion
```

This work sets up the foundation for creating a common framework that mines text to create edges.
Within this framework we would continuously ingest new knowledge as novel findings are published, while providing a single confidence score for an edge via sentence score consolidation.

> Different from existing hetnets like Hetionet where text-derived edges generally cannot be exactly attributed to excerpts from literature [@doi:10.7554/eLife.26726; @doi:10.15363/thinklab.d67], our approach would annotate each edge based on its source sentences.

Reviewer: ```suggestion
As opposed to many existing knowledge graphs, for example Hetionet where text-derived edges generally cannot be exactly attributed to excerpts from literature [@doi:10.7554/eLife.26726; @doi:10.15363/thinklab.d67], our approach has the potential to annotate each edge based on its source sentences.
```

In addition, edges generated with this approach would be unencumbered from upstream licensing or copyright restrictions, enabling openly licensed hetnets at a scale not previously possible [@doi:10.15363/thinklab.d107; @doi:10.1038/536016a; @doi:10.1371/journal.pone.0213090].

> Accordingly, we plan to use this framework to create a robust multi-edge extractor via multitask learning [@doi:10.1145/3209889.3209898] to construct continuously updating literature-derived hetnets.

Reviewer: ```suggestion
New multitask learning [@doi:10.1145/3209889.3209898] strategies may make it even more practical to reuse label functions to construct continuously updating literature-derived knowledge graphs.
```
